Python_DataEngineering/project.py

```diff
@@ -0,0 +1,119 @@
+    result=mycursor.fetchone()
+    print(result)
+    
+    #It will fetch nof of bikes ,as well as at which dockpoint bikes are available according to the timestamp user input's.
+    print("Enter the timestamp to get the available bikes at that moment ")
+    sql_statement1="select nof_of_bikes,dock_point from timestamps_table1 where timestamp= %s and date=%s"
+    timestamps=input()
+    mycursor.execute(sql_statement1,(timestamps,date))
+    bikes_at_dock_points=mycursor.fetchall()
+    for i in bikes_at_dock_points:
+        print(i)
+    
+    
+# This method returns  no of available bikes per particular dockpoint as a key value pair
+#key-docking point,value-[nof of bikes,timestamp,date]
+def bike_availability_at_doctpoint(list_of_json):
+    total_no_of_bikes=0
+    dict_of_available_bikes={}
+    for dicts in list_of_json:
+        for keys in dicts["additionalProperties"]:
+            if keys["key"]=="NbBikes":
+                if dicts["commonName"] not in dict_of_available_bikes:
+                    dict_of_available_bikes[dicts["commonName"]]=[]
+                
+                dict_of_available_bikes[dicts["commonName"]].append(int(keys["value"]))
+                dict_of_available_bikes[dicts["commonName"]].append((keys["modified"][11:-1]))
+                dict_of_available_bikes[dicts["commonName"]].append((keys["modified"][:10]))
+                
+                
+                total_no_of_bikes+=int(keys["value"])
+    return dict_of_available_bikes
+                
+    
+# This method returns nof of bikes available in docking point  in that particular time stamp in form of key value pair.
+#key-timestamp
+#value-[nof_of_bikes,dockingpoint,date]
+def bike_availabilty_at_timestamp(list_of_json):
+    
+    dict_of_timestamps={}
+    for dicts in list_of_json:
+        for keys in dicts["additionalProperties"]:
+            if keys["key"]=="NbBikes":
+                timestamps=keys["modified"][11:-1]
+                if timestamps not in dict_of_timestamps:
+                    dict_of_timestamps[timestamps]=[]
+                    
+                dict_of_timestamps[timestamps].append(int(keys["value"]))
+                dict_of_timestamps[timestamps].append(dicts["commonName"])
+                dict_of_timestamps[timestamps].append(keys["modified"][:10])
+    return dict_of_timestamps
+
+# This method requests the api,and converts into json format and calls the methods to check the nof bikes available in dockpints ,as well as in time stamps.
+#This method aslo calls a method which  inserts these required values into database    
+# This method also calls query_from_users to allow users to fetch from database
+def api():
+    response=requests.get("https://api.tfl.gov.uk/bikepoint")#Extracting  API
+    list_of_json=response.json()#converting into json format
+    dict_of_available_bikes=bike_availability_at_doctpoint(list_of_json)
+    dict_of_timestamps=bike_availabilty_at_timestamp(list_of_json)

```

Comment:
You do not need 2 separate tables for this - in the same table you can query from name and from timestamp separately